packages/shared/src/hooks/useOrganizationList.tsx

```diff
@@ -1,39 +1,248 @@
+    }
+    return userInvitationsData?.data ?? [];
+  }, [triggerInfinite, userInvitationsDataInfinite, userInvitationsData]);
+
+  const isomorphicCount = useMemo(() => {
+    if (triggerInfinite) {
+      return userInvitationsDataInfinite?.[userInvitationsDataInfinite?.length - 1]?.total_count || 0;
+    }
+    return userInvitationsData?.total_count ?? 0;
+  }, [triggerInfinite, userInvitationsDataInfinite, userInvitationsData]);
+
+  const isomorphicIsLoading = triggerInfinite ? userInvitationsLoadingInfinite : userInvitationsLoading;
+  const isomorphicIsFetching = triggerInfinite ? userInvitationsInfiniteValidating : userInvitationsValidating;
+  const isomorphicIsError = !!(triggerInfinite ? userInvitationsInfiniteError : userInvitationsError);
+  /**
+   * Helpers
+   */
+  const fetchNext = useCallback(() => {
+    isomorphicSetPage(n => n + 1);
+  }, [isomorphicSetPage]);
+
+  const fetchPrevious = useCallback(() => {
+    isomorphicSetPage(n => n - 1);
+  }, [isomorphicSetPage]);
+
+  const offsetCount = (initialPageRef.current - 1) * initialPageSizeRef.current;
+
+  const pageCount = Math.ceil((isomorphicCount - offsetCount) / initialPageSizeRef.current);
+  const hasNextPage =
+    isomorphicCount - offsetCount * initialPageSizeRef.current > isomorphicPage * initialPageSizeRef.current;
+  const hasPreviousPage = (isomorphicPage - 1) * initialPageSizeRef.current > offsetCount * initialPageSizeRef.current;
+
+  const unstable__mutate = triggerInfinite ? userInvitationsInfiniteMutate : userInvitationsMutate;
+
   // TODO: Properly check for SSR user values
   if (!clerk.loaded || !user) {
-    return { isLoaded: false, organizationList: undefined, createOrganization: undefined, setActive: undefined };
+    return {
+      isLoaded: false,
+      organizationList: undefined,
+      createOrganization: undefined,
+      setActive: undefined,
+      userInvitations: {
+        data: undefined,
+        count: undefined,
+        isLoading: false,
+        isFetching: false,
+        isError: false,
+        page: undefined,
+        pageCount: undefined,
+        fetchPage: undefined,
+        fetchNext: undefined,
+        fetchPrevious: undefined,
+        hasNextPage: false,
+        hasPreviousPage: false,
+        unstable__mutate: undefined,
+      },
+    };
   }
 
   return {
-    isLoaded: true,
+    isLoaded: canFetch,
     organizationList: createOrganizationList(user.organizationMemberships),
     setActive: clerk.setActive,
     createOrganization: clerk.createOrganization,
+    userInvitations: {
+      data: isomorphicData,
+      count: isomorphicCount,
+      isLoading: isomorphicIsLoading,
+      isFetching: isomorphicIsFetching,
+      isError: isomorphicIsError,
+      page: isomorphicPage,
+      pageCount,
+      fetchPage: isomorphicSetPage,
+      fetchNext,
+      fetchPrevious,
+      hasNextPage,
+      hasPreviousPage,
+      unstable__mutate,
```

Comment:
❓ Is it necessary to export mutate at this point since invitations are handled from ClerkJS/FAPI calls at this point?

Reply:
I'm not sure what you are referring to. Mutate will be used for revalidation, in order to refresh the data on demand